src/System.Xml.XmlDocument/tests/Performance/Perf.XmlNodeList.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Xml;
+using Microsoft.Xunit.Performance;
+
+namespace XmlDocumentTests.XmlNodeListTests
+{
+    public class Perf_XmlNodeList
+    {
+        [Benchmark]
+        public void GetEnumerator()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                // Setup
+                var doc = new XmlDocument();
+                doc.LoadXml("<a><sub1/><sub2/></a>");
+                XmlNodeList list = doc.DocumentElement.ChildNodes;
+
+                // Actual perf testing
+                using (iteration.StartMeasurement())
+                {
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
+                    list.GetEnumerator(); list.GetEnumerator(); list.GetEnumerator();
```

Comment:
In these tests where you're doing GetEnumerator (here and elsewhere), is GetEnumerator really the thing we want to be testing, or is it the cost of iterating via that enumerator, e.g. by using a foreach loop to go through all of the contents?